apollo-core/src/main/java/com/spotify/apollo/core/Service.java

```diff
@@ -78,24 +77,43 @@
    * Starts a new instance of this service that is fully initialized. It will initialize the service
    * using the {@code config} passed as an argument and the environment variables.
    *
-   * @param args   Command-line arguments for the service.
+   * @param args Command-line arguments for the service.
    * @param config Configuration for the service.
    * @return a new instance of this service that is up and running.
-   * @throws ApolloHelpException   if the user wants to show command-line help and not start the
-   *                               application.
-   * @throws ApolloCliException    if something else related to CLI parsing failed.
-   * @throws java.io.IOException   if the application could not start for some other reason.
+   * @throws ApolloHelpException if the user wants to show command-line help and not start the
+   *     application.
+   * @throws ApolloCliException if something else related to CLI parsing failed.
+   * @throws java.io.IOException if the application could not start for some other reason.
    */
   @VisibleForTesting
   Instance start(String[] args, Config config) throws IOException;
 
   /**
-   * A builder for a new service.
+   * Starts a new instance of this service that is fully initialized. It will initialize the service
+   * using the {@code config} passed as an argument and the environment variables and a set of extra
+   * {@link ApolloModule} that should always be loaded.
+   *
+   * @param args Command-line arguments for the service.
+   * @param config Configuration for the service.
+   * @param extraModules Set of modules that should be loaded in addition to the ones loaded by the
+   *     {@link Builder}. These modules will be deduplicated against other modules based on {@link
+   *     ApolloModule#getId()}.
+   * @return a new instance of this service that is up and running.
+   * @throws ApolloHelpException if the user wants to show command-line help and not start the
+   *     application.
+   * @throws ApolloCliException if something else related to CLI parsing failed.
+   * @throws java.io.IOException if the application could not start for some other reason.
    */
+  default Instance start(String[] args, Config config, Set<ApolloModule> extraModules)
+      throws IOException {
+    return this.start(args, config);
```

Comment:
Not sure I understand why this default implementation is required.
This default behavior does not match what the docstring says.
I.e. if nothing overrides this the `extraModules` are ignored

Reply:
Discussion took place in separate communication channel.
I'm fine with whatever :slightly_smiling_face: